activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -100,6 +100,8 @@ def create_binds_for_hash(attributes)
             # For polymorphic relationships, find the foreign key and type:
             # PriceEstimate.where(estimate_of: treasure)
             result[column_name] = AssociationQueryHandler.value_for(table, column_name, value)
+          when value.is_a?(Array) && !table.type(column_name).respond_to?(:subtype)
```

Comment:
> because handlers doesn't support to produce binds

Right.. I think that's what I was saying we should fix, back in #26202.

This is already a pretty tangled mess of conditionals... it seems wiser for us to try to find a way to rearrange it and hammer them out *before* we add more. 😕  (Even though I do really want to see the below test passing.)